forum/forum.py

```diff
@@ -13,345 +13,389 @@
-	links = []
-	subforum = Subforum.query.filter(Subforum.id == subforumid).first()
-	parent = Subforum.query.filter(Subforum.id == subforum.parent_id).first()
-	links.append("<a href=\"/subforum?sub=" + str(subforum.id) + "\">" + subforum.title + "</a>")
-	while parent is not None:
-		links.append("<a href=\"/subforum?sub=" + str(parent.id) + "\">" + parent.title + "</a>")
-		parent = Subforum.query.filter(Subforum.id == parent.parent_id).first()
-	links.append("<a href=\"/\">Forum Index</a>")
-	link = ""
-	for l in reversed(links):
-		link = link + " / " + l
-	return link
+    links = []
+    subforum = Subforum.query.filter(Subforum.id == subforumid).first()
+    parent = Subforum.query.filter(Subforum.id == subforum.parent_id).first()
+    links.append("<a href=\"/subforum?sub=" + str(subforum.id) + "\">" + subforum.title + "</a>")
+    while parent is not None:
+        links.append("<a href=\"/subforum?sub=" + str(parent.id) + "\">" + parent.title + "</a>")
+        parent = Subforum.query.filter(Subforum.id == parent.parent_id).first()
+    links.append("<a href=\"/\">Forum Index</a>")
+    link = ""
+    for l in reversed(links):
+        link = link + " / " + l
+    return link
 
 
-#from forum.app import db, app 
+# from forum.app import db, app
 
 
 login_manager = LoginManager()
 login_manager.init_app(app)
 
+
 # if __name__ == "__main__":
 # 	#runsetup()
 # 	port = int(os.environ["PORT"])
 # 	app.run(host='0.0.0.0', port=port, debug=True)
 
 
-
-#DATABASE STUFF
+# DATABASE STUFF
 @login_manager.user_loader
 def load_user(userid):
-	return User.query.get(userid)
+    return User.query.get(userid)
 
 
 password_regex = re.compile("^[a-zA-Z0-9!@#%&]{6,40}$")
 username_regex = re.compile("^[a-zA-Z0-9!@#%&]{4,40}$")
-#Account checks
+
+
+# Account checks
 def username_taken(username):
-	return User.query.filter(User.username == username).first()
+    return User.query.filter(User.username == username).first()
+
+
 def email_taken(email):
-	return User.query.filter(User.email == email).first()
+    return User.query.filter(User.email == email).first()
+
+
 def valid_username(username):
-	if not username_regex.match(username):
-		#username does not meet password reqirements
-		return False
-	#username is not taken and does meet the password requirements
-	return True
+    if not username_regex.match(username):
+        # username does not meet password reqirements
+        return False
+    # username is not taken and does meet the password requirements
+    return True
+
+
 def valid_password(password):
-	return password_regex.match(password)
-#Post checks
+    return password_regex.match(password)
+
+
+# Post checks
 def valid_title(title):
-	return len(title) > 4 and len(title) < 140
+    return len(title) > 4 and len(title) < 140
+
+
 def valid_content(content):
-	return len(content) > 10 and len(content) < 5000
+    return len(content) > 10 and len(content) < 5000
 
 
-#OBJECT MODELS
+# OBJECT MODELS
 class User(UserMixin, db.Model):
```

Comment:
First real change here, removed unique from "admin" and added a "picture" column

Reply:
This is the only other real change aside from the ones at the top